contrib/pkg/clusterpool/clusterpool.go

```diff
@@ -0,0 +1,419 @@
+				Region:         o.Region,
+			}
+		}
+	}
+
+	imageSet, err := o.configureImages()
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := o.build(builder)
+	if err != nil {
+		return nil, err
+	}
+
+	if imageSet != nil {
+		result = append(result, imageSet)
+	}
+	if len(pullSecret) != 0 {
+		result = append(result, builder.GeneratePullSecretSecret())
+	}
+
+	return result, nil
+}
+
+func (o *ClusterPoolOptions) configureImages() (*hivev1.ClusterImageSet, error) {
+	if len(o.ClusterImageSet) > 0 {
+		return nil, nil
+	}
+
+	if o.ReleaseImage == "" {
+		if o.ReleaseImageSource == "" {
+			return nil, fmt.Errorf("specify either a release image or a release image source")
+		}
+		var err error
+		o.ReleaseImage, err = utils.DetermineReleaseImageFromSource(o.ReleaseImageSource)
+		if err != nil {
+			return nil, fmt.Errorf("cannot determine release image: %v", err)
+		}
+	}
+	imageSet := &hivev1.ClusterImageSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-imageset", o.Name),
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterImageSet",
+			APIVersion: hivev1.SchemeGroupVersion.String(),
+		},
+		Spec: hivev1.ClusterImageSetSpec{
+			ReleaseImage: o.ReleaseImage,
+		},
+	}
+	o.ClusterImageSet = imageSet.Name
+	return imageSet, nil
+}
+
+func (o *ClusterPoolOptions) build(builder *clusterresource.Builder) ([]runtime.Object, error) {
+
+	var allObjects []runtime.Object
+
+	allObjects = append(allObjects, o.generateClusterPool())
+
+	if o.createCloudSecret {
+		cloudCredsSecret := builder.CloudBuilder.GenerateCredentialsSecret(builder)
+		if cloudCredsSecret != nil {
+			allObjects = append(allObjects, cloudCredsSecret)
+		}
+	}
+
+	return allObjects, nil
+}
+
+func (o *ClusterPoolOptions) getPullSecretSecretName() string {
+	return fmt.Sprintf("%s-pull-secret", o.Name)
+}
+
+func (o *ClusterPoolOptions) addClusterPoolPlatform(cp *hivev1.ClusterPool) {
+	switch o.Cloud {
+	case cloudAWS:
+		cp.Spec.Platform = hivev1.Platform{
+			AWS: &hivev1aws.Platform{
+				CredentialsSecretRef: corev1.LocalObjectReference{
+					Name: o.credsSecretName(),
+				},
+				Region: o.Region,
+			},
+		}
+	case cloudAzure:
+		cp.Spec.Platform = hivev1.Platform{
+			Azure: &hivev1azure.Platform{
+				CredentialsSecretRef: corev1.LocalObjectReference{
+					Name: o.credsSecretName(),
+				},
+				Region:                      o.Region,
+				BaseDomainResourceGroupName: o.AzureBaseDomainResourceGroupName,
+			},
+		}
+	case cloudGCP:
+		cp.Spec.Platform = hivev1.Platform{
+			GCP: &hivev1gcp.Platform{
+				CredentialsSecretRef: corev1.LocalObjectReference{
+					Name: o.credsSecretName(),
+				},
+				Region: o.Region,
+			},
+		}
+	}
+}
+
+func (o *ClusterPoolOptions) credsSecretName() string {
+	if o.createCloudSecret {
+		switch o.Cloud {
+		case cloudAWS:
+			return fmt.Sprintf("%s-aws-creds", o.Name)
+		case cloudAzure:
+			return fmt.Sprintf("%s-azure-creds", o.Name)
+		case cloudGCP:
+			return fmt.Sprintf("%s-gcp-creds", o.Name)
```

Comment:
This knowledge is already known in the CloudBuilder. We should not need to take it out from there. Add a function to the CloudBuilder that will return the creds secret name.